src/battle_ai_util.c

```diff
@@ -820,47 +820,47 @@ s32 AI_CalcDamage(u16 move, u8 battlerAtk, u8 battlerDef, u8 *typeEffectiveness,
         else
             dmg = (critDmg + normalDmg * (critChance - 1)) / critChance;
 
-        // Handle dynamic move damage
-        switch (gBattleMoves[move].effect)
+        if (!considerZPower)
         {
-        case EFFECT_LEVEL_DAMAGE:
-        case EFFECT_PSYWAVE:
-            dmg = gBattleMons[battlerAtk].level * (AI_DATA->abilities[battlerAtk] == ABILITY_PARENTAL_BOND ? 2 : 1);
-            break;
-        case EFFECT_DRAGON_RAGE:
-            dmg = 40 * (AI_DATA->abilities[battlerAtk] == ABILITY_PARENTAL_BOND ? 2 : 1);
-            break;
-        case EFFECT_SONICBOOM:
-            dmg = 20 * (AI_DATA->abilities[battlerAtk] == ABILITY_PARENTAL_BOND ? 2 : 1);
-            break;
-        case EFFECT_MULTI_HIT:
-            dmg *= (AI_DATA->abilities[battlerAtk] == ABILITY_SKILL_LINK ? 5 : 3);
-            break;
-        case EFFECT_TRIPLE_KICK:
-            dmg *= (AI_DATA->abilities[battlerAtk] == ABILITY_SKILL_LINK ? 6 : 5);
```

Comment:
This line was not a mistake. `AI_CalcDamage` is only modifying damage for the AI's calculations; it is not applied to the actual damage done by the move. Without this the AI will treat Triple Kick and Triple Axel as 10 and 20 power moves, respectively.

If all hits connect then the base power is 10 + 20 + 30 = 60 for Triple Kick and 20 + 40 + 60 = 120 for Triple Axel; 6x the base power of the move. Without Skill Link, the average damage is less due to the chance to miss on every hit (hence the x5 multiplier).

Reply:
Yeah, I know that it isn't applied to the actual damage but the moves with `EFFECT_TRIPLE_KICK` still got a really high damage calc and was often used unreasonably by the AI when considering the moves.

The damage was is so high because when `AI_CalcDamage` Calcs the damage for `EFFECT_TRIPLE_KICK` `CalculateMoveDamageAndEffectiveness` is called. Where it got a BP of 40 for Triple Kick and 80 for Triple Axel. And on top of that the dynamic damage was applied.
```
[INFO] GBA Debug:	First calc 28
[INFO] GBA Debug:	Second calc 168
```

> Without this the AI will treat Triple Kick and Triple Axel as 10 and 20 power moves, respectively.

```
        if (gMultiHitCounter == 0)
            basePower *= 6;
        else
            basePower *= (4 - gMultiHitCounter);
```
I changed it so it still considered but only it would hit all 3.

The reason I did it like that is because I think that since it has a 70% of hitting all, it should consider the moves as max BP. Though I can rewrite it an a way that `CalculateMoveDamageAndEffectiveness` isn't multiplying the base power when it is not applied to the damage calc and keep the Skill Link calcs..

Reply:
Ahh yep my bad, you're right. Though I'd suggest adding a comment above `if (gMultiHitCounter == 0)` to explain that that's just for the AI, to avoid confusion.

Reply:
Sorry for the delay! I think just the comment is fine. We don't account for accuracy anywhere else in the AI's damage calculations, so I think it's fine to remove the Skill Link check.

Reply:
Done, thanks!